book/src/validity-predicates.md

```diff
@@ -1,122 +1,61 @@
+Some simple VP examples:
+- white list VP allows only specific users to change the application state
+- lower bound VP restricts the smallest amount of asset that can be received by a user
 
-#### State model in Taiga
+### Hierarchical VP structure
+Validity predicates in Taiga have a hierarchical structure. For all applications involved in a transaction, their `ApplicationVP` must be checked. 
+These VPs might require validity of some other VPs in order for a transaction to be considered valid. 
+In that case, those other VPs must be checked too.
 
-Unlike transparent Anoma, which operates on an account model, Taiga uses a note based model. In an account model, each account has associated state which is stored in a database, and updated according to that account's validity predicate. In the note model, however, state is sharded into an append-only set of *note commitments* and revealed *nullifiers*. Each note has exactly one associated nullifier, and the current state of the Taiga is the subset of notes whose nullifiers are not yet revealed, called unspent notes. The Action/Execute circuit, together with the transaction verifier, ensure that the Taiga state is consistent.
+Hierarchical structure of VPs might come in handy in the situations where the application wants to enforce different checks depending on the context. 
+One example of it could be applications that have users and allow them to have user-defined VPs. 
+The application VP then will enforce checking the `userVP` of each user involved in the transaction.
 
-### PLONK circuit configuration for validity predicates in Taiga
+![img.png](images/vp_hierarchy.png)
 
-The validity predicate configuration includes the following "gates" in the PLONK configuration:
+### Transparent vs shielded VPs
+Validity predicates in principle can be both transparent and shielded. Transparent VPs are represented as WASM code and publicly visible, 
+shielded VPs are represented as arithmetic circuits hidden under ZKPs. Each transaction in Taiga has VP proofs attached to it, 
+and whoever has the verifying key (VK), can verify the proofs.
 
-* Field addition/multiplication
-* Elliptic curve addition and scalar multiplication
-* Poseidon hash
+![img.png](images/vp_img.png)
 
-#### Addresses
+Taiga uses the Halo2 proving system and validity predicates are represented as [PLONKish circuits](https://zcash.github.io/halo2/concepts/arithmetization.html). 
+For privacy reasons, all Taiga VPs share the same configuration (the set of building blocks available for the circuits), 
+and distinct VPs are created by specifying the *selectors* (that select which blocks to use).
```

Comment:
> For privacy reasons, all Taiga VPs share the same configuration (the set of building blocks available for the circuits), 
> and distinct VPs are created by specifying the *selectors* (that select which blocks to use).

I don't think this is true; I think each VP can have a different configuration.
E.g.
```
#[derive(Clone, Debug)]
pub struct SudokuVPConfig {
    note_config: NoteConfig,
}
```
but we can add any other configurations if needed

Reply:
Ah, it is a piece of the old document and I guessed it wouldn't change. Maybe I'll just remove it as it seems to be a very low-level detail as well